### ISSUE TITLE
Decommissioning PAPR

### ISSUE THREAD

**Issue by jlebon** [jlebon/redhat-ci#105]
TL;DR: We're planning to decommission PAPR in the near future (targeting end of Dec 2019). Please start looking for alternative CI solutions.

---

The reasons for this boil down to:
1. we don't have enough cycles to maintain it and push it forward
2. the infrastructure it relies on has been (and still is) flaky
3. it doesn't align well with other FCOS/RHCOS projects and the rest of OpenShift

So what are the alternatives?
- if you like (or don't mind) better alignment with OpenShift, you can hook into the Prow instance maintained at https://github.com/openshift/release
- if your tests require no or few privileges, Travis CI may be fine
- if you need more privileges/customization, other third-party services like Cirrus CI may work
- CentOS CI is also a valid alternative, though requires more elbow grease (there's a public Jenkins instance, bare-metal (Duffy) nodes, as well as an OpenShift cluster): https://wiki.centos.org/QaWiki/CI/GettingStarted

I know many of us have already started work on adopting those alternatives. For example, libpod has moved to Cirrus CI + Prow, and there is work in progress to migrate buildah as well.

We've also been experimenting in a few repos (notably rpm-ostree and ostree) with using Prow and CentOS CI (see [2] related to this). We're not there yet, but regardless of the outcome, I don't expect we'll be staying with PAPR much longer.

That said, I *don't* want anyone to be left behind without a replacement CI solution in place. If you need help evaluating options, we can likely provide input (and possibly help out). With that in mind, let's put a soft target of end-of-year 2019?

[1] https://github.com/projectatomic/papr
[2] https://github.com/coreos/fedora-coreos-tracker/issues/263

**Comment by cgwalters** [comment]
See also:

 - https://github.com/containers/buildah/issues/1915
 - https://github.com/containers/bubblewrap/issues/335

**Comment by jlebon** [comment]
Ahh very nice, looks like flatpak has already moved to GitHub actions:
- https://github.com/flatpak/flatpak/commit/266e12e432bda2fb2bf2175ec0ceb8a0a986ed18
- https://github.com/flatpak/flatpak/commit/c4e6c32f6857c216b843c09a2a5dfc7984446034

And similarly for flatpak-builder.

So I think this just leaves:

- [x] rpm-ostree
- [x] ostree
- [x] buildah
- [ ] bubblewrap
- [x] atomic-host-tests (changes there should be rare at this point; can probably just rely on manual testing?)

**Comment by miabbott** [comment]
We've done a poor job of keeping `atomic-host-test` working with PAPR and the project is in maintenance mode, so I'll likely just remove the integration in the near future.

**Comment by jlebon** [comment]
We're finally down to just bubblewrap before we can do this. See:
https://github.com/containers/bubblewrap/issues/335
https://github.com/containers/bubblewrap/issues/348

**Comment by jlebon** [comment]
I deleted the OpenStack PAPR workers today. Updated the README.md with links to CoreOS CI and Prow and the tracker issue. Archiving repo now.

So long PAPR and thanks for all the CI!
